### PR TITLE
HTML5 recording of cameras and screenshare

### PR DESCRIPTION
**Pull request by lfzawacki**
This PR introduces the concept of recording the webrtc cameras and screen shared via the HTML5 client. These are recorded server side by the bbb-webrtc-sfu component using the media server and can be configured in the `bigbluebutton/labs/bbb-webrtc-sfu/config/default.yml` file. For now, the only parameters are these ones:

    recordWebcams: true
    recordScreenSharing: true
    recordingBasePath: "file:///var/kurento/recordings/"

The bbb-webrtc-sfu component will also write the `StartWebRTCShareEvent` and `StopWebRTCShareEvent` to redis so that it can be picked up by the recording processor. Here's two example of the events:

    1524593447.136004 [0 127.0.0.1:48184] "hmset" "recording:183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524582934682:339" "eventName" "StartWebRTCShareEvent" "module" "bbb-webrtc-sfu" "meeting_id" "183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524582934682" "timestamp" "3020619620" "timestampUTC" "1524593447124" "filename" "file:///var/kurento/recordings//183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524582934682/medium-w_iidfyt4ol2gl-1524593447033.mp4"

    1524594012.579308 [0 127.0.0.1:48204] "hmset" "recording:183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524582934682:340" "eventName" "StopWebRTCShareEvent" "module" "bbb-webrtc-sfu" "meeting_id" "183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524582934682" "timestamp" "3021185052" "timestampUTC" "1524594012557" "filename" "file:///var/kurento/recordings//183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524582934682/medium-w_iidfyt4ol2gl-1524593447033.mp4"

### CODE REVIEW

**Comment by lfzawacki** [comment]
Now successfully tested. Just missing the screenshare event to properly close this.

**Comment by ffdixon** [comment]
:-)

**Comment by lfzawacki** [comment]
Examples of the start and stop events for the screenshare. Notice they're sent by different modules.

    1524666091.389988 [0 127.0.0.1:53942] "hmset" "recording:183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524666066757:1848" "eventName" "StartWebRTCDesktopShareEvent" "module" "bbb-webrtc-sfu" "meeting_id" "183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524666066757" "timestamp" "3093263869" "timestampUTC" "1524666091373" "filename" "file:///var/kurento/recordings//183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524666066757-SCREENSHARE/73617-SCREENSHARE-1524666091208.mp4"

    1524666110.043588 [0 127.0.0.1:53948] "HMSET" "recording:183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524666066757:1850" "stream" "sip_73617-SCREENSHARE_1524666091254" "eventName" "StopWebRTCDesktopShareEvent" "duration" "11" "meetingId" "183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1524666066757" "timestamp" "3093282528" "module" "WEBRTC-DESKSHARE"

File: labs/bbb-webrtc-sfu/config/default.example.yml [hunk 1]
```diff
@@ -18,6 +18,10 @@ to-akka: "to-akka-apps-redis-channel"
 webcam-force-h264: true
 screenshare-force-h264: true
 
+recordScreenSharing: true
+recordWebcams: true
```
Comment (prlanzarin) [inline labs/bbb-webrtc-sfu/config/default.example.yml]: We'll be leaving them true by default?

Comment (lfzawacki) [reply labs/bbb-webrtc-sfu/config/default.example.yml]: It's the example file, but I think you're right.... we can probably leave them disabled by default

Comment (kepstin) [reply labs/bbb-webrtc-sfu/config/default.example.yml]: Yes, this should be true by default - but the sfu should not record video to files for meetings that do not have recording enabled

**Comment by kepstin** [comment]
I think there's one remaining thing to do - adjust the cron job (`/etc/cron.daily/bigbluebutton`) so it deletes old files from /var/kurento/recordings and /var/kurento/screenshare

**Pull request merged**